packages/credentials/src/compact-jwt.ts

```diff
@@ -0,0 +1,229 @@
+    }
+
+    const { signer, options } = CompactJwt.algorithms[algorithmId];
+
+    const signatureBytes = await signer.sign({ key: privateKeyJwk as PrivateKeyJwk, data: toSignBytes, algorithm: options! });
+    const base64UrlEncodedSignature = Convert.uint8Array(signatureBytes).toBase64Url();
+
+    return `${toSign}.${base64UrlEncodedSignature}`;
+  }
+
+  /**
+   * Verifies a JWT.
+   * @param params - Parameters for JWT verification
+   * @returns Verified JWT information including signer DID, header, and payload.
+   * @example
+   * ```
+   * const verifiedJwt = await CompactJwt.verify({ compactJwt: myJwt });
+   * ```
+   */
+  static async verify(params: VerifyJwtParams) {
+    const { decoded: decodedJwt, encoded: encodedJwt } = CompactJwt.decode({ compactJwt: params.compactJwt });
+    // TODO: should really be looking for verificationMethod with authentication verification relationship
+    const verificationMethod = await CompactJwt.didResolver.dereference({ didUrl: decodedJwt.header.kid! });
+    if (!utils.isVerificationMethod(verificationMethod)) { // ensure that appropriate verification method was found
+      throw new Error('Verification failed: Expected kid in JWT header to dereference a DID Document Verification Method');
+    }
+
+    // will be used to verify signature
+    const publicKeyJwk = verificationMethod.publicKeyJwk as JwkParamsEcPrivate | JwkParamsOkpPrivate;
+    if (!publicKeyJwk) { // ensure that Verification Method includes public key as a JWK.
+      throw new Error('Verification failed: Expected kid in JWT header to dereference to a DID Document Verification Method with publicKeyJwk');
+    }
+
+    const signedData = `${encodedJwt.header}.${encodedJwt.payload}`;
+    const signedDataBytes = Convert.string(signedData).toUint8Array();
+
+    const signatureBytes = Convert.base64Url(encodedJwt.signature).toUint8Array();
+
+    const algorithmId = `${decodedJwt.header.alg}:${publicKeyJwk['crv'] || ''}`;
+    if (!(algorithmId in CompactJwt.algorithms)) {
+      throw new Error(`Verification failed: ${algorithmId} not supported`);
+    }
+
+    const { signer, options } = CompactJwt.algorithms[algorithmId];
+
+    const isLegit = await signer.verify({
+      algorithm : options!,
+      key       : publicKeyJwk as PublicKeyJwk,
+      data      : signedDataBytes,
+      signature : signatureBytes
+    });
+
+    if (!isLegit) {
+      throw new Error('Signature verification failed: Integrity mismatch');
+    }
+
+    return {
+      signerDid : verificationMethod.controller,
+      header    : encodedJwt.header,
+      payload   : encodedJwt.payload,
+    };
+  }
+
+  /**
+   * Decodes a JWT without verifying its signature.
+   * @param params - Parameters for JWT decoding, including the JWT string.
+   * @returns Decoded JWT parts, including header and payload.
+   * @example
+   * const decodedJwt = CompactJwt.decode({ compactJwt: myJwt });
+   */
+  static decode(params: DecodeJwtParams) {
```

Comment:
thinking it might make more sense to name this `parse` instead of `decode`. thoughts?

Reply:
I think decode is fine:
we use parse as taking the jwt and turning it into a vc:
```
parseJwt(vcJwt: string): VerifiableCredential 
```

so this is actually decoding like jwt.io yea?

Reply:
I like decode here

Reply:
This seems to return `decoded` AND `encoded` parts of the JWK. In it's current form, `parse` seems to make more sense, because it's not just decoding.

Reply:
What's the use case for returning the `encoded` pieces? Since they're in the input `params` we're just splitting apart and then re-returning

Reply:
great q @KendallWeihe. both encoded and decoded jwt's are used in `verify` [here](https://github.com/TBD54566975/web5-js/blob/verify-jwt/packages/credentials/src/compact-jwt.ts#L136-L150)

Reply:
Yeah okay, I'm on team parse now

Reply:
I mean the alternative is to split this into two methods, `decode` and maybe `expand` (or maybe `CompactJwt.spread()`?) there is not good antonym to "compact" ... so yeah, I'm team parse lol

Reply:
updated to use `parse`